### DISCUSSION TITLE
Setup with Cloudflare

### DISCUSSION THREAD

**Discussion by mahmoodn** [XTLS/Xray-core#3790] [Q&A]
My domain name is proxied by Cloudflare and the web panel is accessible through one of the recognized ports. On the client side with VLESS+TLS configuration, when I specify the server's IP address, I am able to ping the server, however, when I specify the domain name, the server is not pingable.

Is there are specific settings for xray for Cloudflare?

**Comment by meower1** [comment]
If you're trying to put your vless configuration behind cloudflare you should use transmission protocols that support cdn. 
example: vless + (websocket/grpc) + TLS

**Comment by mahmoodn** [comment]
OK I did that but apparently reaching out the server via vless + (websocket/grpc) + TLS is not possible. More information is given below:
1- The server IP is 104.X.Y.106
2- The domain name is DM.ORG
3- In the Cloudflare website, I have added two records:
<img width="577" alt="image" src="https://github.com/user-attachments/assets/12834060-c2a1-4362-b42b-ee4ffd7901c6">
4- Using the online DNS lookup, I see two different IPs which are Cloudflare proxies:
<img width="400" alt="image" src="https://github.com/user-attachments/assets/4cbcfeee-d4ba-478d-8758-6dc4a7741d93">
5- On the server side, I have installed the SSL issued from Cloudflare. With respect to the supported ports mentioned [here](https://developers.cloudflare.com/fundamentals/reference/network-ports/#:~:text=Learn%20which%20network%20ports%20Cloudflare%20proxies%20by%20default%20and%20how), I have set the web panel to 8443. I see private and public keys in the installation folder `DM.ORG.key` and `DM.ORG.cer`.
6- Now I can access the web panel through https://DM.ORG:8443
7- I have created a a vless+websocket+tls on port 20007 (don't know if it must be in the Cloudflare supported ports or not) and on the client, the ping doesn't work. For example, V2rayN output is shown below:
```
Initial Configuration
Start service (2024/09/13 12:23:01)...
C:\Users\m\Downloads\v2rayN-With-Core\v2rayN-With-Core\guiConfigs\configSpeedtest.json
Xray 1.8.16 (Xray, Penetrates Everything.) 9432a60 (go1.22.4 windows/amd64)
A unified platform for anti-censorship.
2024/09/13 12:23:01 [Info] infra/conf/serial: Reading config: configSpeedtest.json
2024/09/13 12:23:01 [Warning] core: Xray 1.8.16 started
2024/09/13 12:23:02 127.0.0.1:53442 accepted //www.google.com:443 [http10829 -> proxy10829]
2024/09/13 12:23:06 The ping of current service: -1 ms
2024/09/13 12:23:14 [Warning] [3189665587] app/proxyman/outbound: failed to process outbound traffic > proxy/vless/outbound: failed to find an available destination > common/retry: [transport/internet/websocket: failed to dial WebSocket > transport/internet/websocket: failed to dial to (wss://DM.ORG:20007/):  > dial tcp 172.67.185.182:20007: i/o timeout transport/internet/websocket: failed to dial WebSocket > transport/internet/websocket: failed to dial to (wss://DM.ORG:20007/):  > dial tcp 104.21.19.92:20007: operation was canceled transport/internet/websocket: failed to dial WebSocket > transport/internet/websocket: failed to dial to (wss://DM.ORG:20007/):  > dial tcp: lookup DM.ORG: operation was canceled] > common/retry: all retry attempts failed

```
As you can see it is dialing TCP. I don't know if it is related to TLS handshaking or port number or anything else. In fact, when I ping the server, I don't see a message on the server side. So, on the server side it means nothing has received.

Do you have any idea about that?

**Comment by AlbertMingXu** [comment]
I have the same question with you.

**Comment by wpdevelopment11** [comment]
Maybe [this](https://github.com/wpdevelopment11/xray-tutorial?tab=readme-ov-file#step-4---cloudflare-proxy-optional) will be helpful.

**Comment by Benjamin1919** [comment]
You can only use ports supported by Cloudflare: https://developers.cloudflare.com/fundamentals/reference/network-ports/
Apparently, 20007 is not in the list.

**Comment by vltmk** [comment]
**Here's the [problem](https://community.cloudflare.com/t/blocking-xray-and-v2ray-traffic/748125):**
"Cloudflare updated the Terms of Service on Dec 3, 2024 to explicitly call out that proxy services, like a VPN service, are prohibited from running on Cloudflare’s network unless explicitly approved."
